### PR TITLE
RATEST-81: Fix Using back button in merge patient test

### PR DESCRIPTION
**Pull request by Irenyak1**
Ticket [ID](https://issues.openmrs.org/browse/RATEST-81)

Description -> Fix Using back button in merge patient test

### CODE REVIEW

File: ui-tests/src/test/java/org/openmrs/reference/DeleteFormTest.java [hunk 1]
```diff
@@ -15,6 +16,7 @@
 /**
  * Created by nata on 24.06.15.
  */
+@Ignore
```
Comment (Irenyak1) [inline ui-tests/src/test/java/org/openmrs/reference/DeleteFormTest.java:19]: Ignored the test until investigated.

Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/DeleteFormTest.java:19]: Was this change necessary ?

Comment (Irenyak1) [inline ui-tests/src/test/java/org/openmrs/reference/DeleteFormTest.java:19]: Ignored the test until investigated.

File: ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java [hunk 2]
```diff
@@ -30,23 +30,20 @@
 
     @Before
     public void setUp() throws Exception {
-       
         homePage = new HomePage(page);
-        assertPage(homePage);
+        assertPage(homePage.waitForPage());
```
Comment (Irenyak1) [inline ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java:34]: During the test setUp(), many pages are invoked which takes some time for the patient registration page to be loaded by the classLoader. Therefore adding waitForPage() caters for the time taken to load the resource such that the following operations carry on smoothly.

File: ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
 
 import org.junit.Before;
+import org.junit.After;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.openmrs.reference.groups.BuildTests;
 import org.openmrs.reference.helper.TestPatient;
 import org.openmrs.reference.page.*;
 import org.openmrs.uitestframework.test.TestBase;
-import org.junit.*;
```
Comment (Irenyak1) [inline ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java:8]: This was calling the entire package which is not necessary since we need only a particular tag.

File: ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java [hunk 4]
```diff
@@ -76,15 +73,12 @@ public void usingBackButtonInMergePatientTest() throws Exception {
...
 
-
     @After
     public void tearDown() throws Exception {
         headerPage.clickOnHomeIcon();
         TestData.PatientInfo p = new TestData.PatientInfo();
         p.uuid = patient.uuid;
         deletePatient(p);
         waitForPatientDeletion(patient.uuid);
-        headerPage.logOut();
```
Comment (Irenyak1) [inline ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java:87]: There's no need to invoke the headerPage.logOut() in the tearDown() function after the test patient has been deleted from the database. The test still passes without this function.

File: ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java [hunk 5]
```diff
@@ -1,23 +1,23 @@
 package org.openmrs.reference;
 
 import org.junit.Before;
+import org.junit.After;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.openmrs.reference.groups.BuildTests;
 import org.openmrs.reference.helper.TestPatient;
 import org.openmrs.reference.page.*;
```
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java:9]: explicitly import classes since this would avail all pages which is not right

File: ui-tests/src/test/java/org/openmrs/reference/DeleteFormTest.java (hunk 1 above)
Comment (kdaud) [reply ui-tests/src/test/java/org/openmrs/reference/DeleteFormTest.java]: > Ignored the test until investigated.

Any reason for this comment? I can see that specific class is already ignore.

File: ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java (hunk 2 above)
Comment (kdaud) [reply ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java:34]: > Therefore adding waitForPage() caters for the time taken to load the resource such that the following operations carry on smoothly.

Ideally is this true for the change introduced at code `LN-34` ?

File: ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java (hunk 5 above)
Comment (Irenyak1) [reply ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java:9]: @kdaud I have not understood your comment.

Comment (Irenyak1) [reply ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java:9]: In view of the code pointed to.

**Comment by kdaud** [comment]
Well-done @Irenyak1 
Please catch my reviews above!

File: ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java (hunk 5 above)
Comment (kdaud) [reply ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java:9]: > I have not understood your comment.

Check [here](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/610b3dbcba58f081d13ea0e842994f852477ff8c/ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java#L9)

File: ui-tests/src/test/java/org/openmrs/reference/DeleteFormTest.java (hunk 1 above)
Comment (Irenyak1) [reply ui-tests/src/test/java/org/openmrs/reference/DeleteFormTest.java]: @kdaud that is an outdated comment like you can see

File: ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java (hunk 5 above)
Comment (Irenyak1) [reply ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java]: @kdaud worked on it already

**Comment by kdaud** [comment]
Please fix the build failure

File: ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java (hunk 2 above)
Comment (Irenyak1) [reply ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java:38]: Kinda, because without waiting for the homePage to load, the assertion will return null. So yeah the waitForPage(), works to await the loading of the so that the assertPage returns true. @kdaud .
